recipes/wasmtime-cpp/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+find_package(wasmtime-cpp REQUIRED)
+
+add_executable(example example.cpp)
+target_link_libraries(example PRIVATE wasmtime-cpp::wasmtime-cpp)
+target_compile_definitions(example PUBLIC WASMTIME_EXAMPLES_PATH="${CMAKE_SOURCE_DIR}")
+target_compile_options(example PRIVATE ${CONAN_COMPILE_DEFINITIONS_WASMTIME})
```

Comment:
After #7190
```suggestion
include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(wasmtime-cpp REQUIRED)

add_executable(example example.cpp)
target_link_libraries(example PRIVATE wasmtime-cpp::wasmtime-cpp)
target_compile_definitions(example PRIVATE WASMTIME_EXAMPLES_PATH="${CMAKE_SOURCE_DIR}")
set_property(TARGET example PROPERTY CXX_STANDARD 17)
```

Reply:
Disagree ... it just dependent on who used to what

Reply:
No need for `CONAN_COMPILE_DEFINITIONS_WASMTIME`.
Conan v2 will drop these variables.

Reply:
Oh it was due to `CONAN_COMPILE_DEFINITIONS_WASMTIME` ... did not get ...
Yes, it is not needed due to proper package dependencies `wasmtime-cpp::wasmtime-cpp`

Reply:
Done

Reply:
@madebr Reverted `CONAN_COMPILE_DEFINITIONS_WASMTIME` as it broke build on Windows

Reply:
You need the modification done in `package_info` of wasmtime in #7190 for it to work.

Reply:
I will need also provide option `shared` ...
But how to delegate it to `wasmtime` package ?

Reply:
wasmtime-cpp is header-only so cannot be built as static/shared library.
When you want to switch the "sharedness" of wasmtime, you need to pass `-o wasmtime:shared=True/False` to conan.

Reply:
Oh ... gotcha, thanks !!

Reply:
@madebr One more question ...
If I will do like this:
```python
    def package_info(self):
        if not self.options.shared:
            if self.settings.os == "Windows":
                self.cpp_info.defines= ["WASM_API_EXTERN=", "WASI_API_EXTERN="]

```
Where to get `self.options.shared` if I do not have such option ?

Reply:
@madebr  Also this addtion only needed because `conan` did not add defines in `target_compile_definitions(...)`
Otherwise I would not need this `package_info` method ...

Why `conan` do not add `target_compile_definitions(...)` ?

Reply:
> @madebr One more question ...
> If I will do like this:
> 
> ```python
>     def package_info(self):
>         if not self.options.shared:
>             if self.settings.os == "Windows":
>                 self.cpp_info.defines= ["WASM_API_EXTERN=", "WASI_API_EXTERN="]
> ```
> 
> Where to get `self.options.shared` if I do not have such option ?

You don't need to do this.
When linking to the target `wasmtime-cpp::wasmtime-cpp`,
you will also automatically link to `wasmtime::wasmtime`.
When you link to `wasmtime::wasmtime`, these defines are automatically added (when needed).

Everything that is done in wasmtime's package_info should be abstracted away.
So apart from linking to `wasmtime-cpp::wasmtime-cpp`, you should not worry about any definitions related to `wasmtime`.

Reply:
Try analyzing the automatically generated `Findwasmtime.cmake` in the `test_package/build/xxxxx` folders.

Reply:
Yeah, fixed